### PR TITLE
ArticlesController#feed refactor

### PR DESCRIPTION
**Pull request by bradpauly**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description
Minor refactor of articles controller feed to reduce knowledge by controller. This adds a new scope and a couple of tests to verify records returned.

## Related Tickets & Documents
https://github.com/thepracticaldev/dev.to/issues/3739

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

File: app/models/article.rb [hunk 1]
```diff
@@ -241,6 +241,12 @@ def self.active_eli5(time_ago)
     stories.pluck(:path, :title, :comments_count, :created_at)
   end
 
+  def self.feed
+    published.
+      select(:id, :published_at, :processed_html, :user_id, :organization_id, :title, :path).
+      order(published_at: :desc)
+  end
+
```
Comment (rhymes) [inline app/models/article.rb]: code this become an actual scope? see for example https://github.com/thepracticaldev/dev.to/blob/0d2506f9ef79a32f7474001337161314d1062134/app/models/article.rb#L79

I'm a little conflicted about adding ordering to scopes though, what do you think @lightalloy ?

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -213,11 +209,8 @@ def handle_user_or_organization_feed
   end
 
   def handle_tag_feed
-    tag = Tag.find_by(name: params[:tag].downcase)
-
-    return unless tag
+    return unless (@tag = Tag.aliased_name(params[:tag]))
```
Comment (rhymes) [inline app/controllers/articles_controller.rb]: I'm not sure saving one line here is fruitful for readability, I'd split this into:

```ruby
@tag = ...
return unless @tag
```

File: app/models/tag.rb [hunk 3]
```diff
@@ -61,6 +61,12 @@ def self.valid_categories
     ALLOWED_CATEGORIES
   end
 
+  def self.aliased_name(word)
+    return unless (tag = find_by(name: word.downcase))
```
Comment (rhymes) [inline app/models/tag.rb]: I'm not sure saving one line here is fruitful for readability, I'd split this into:

```ruby
tag = ...
return unless tag
```

File: spec/models/article_spec.rb [hunk 4]
```diff
@@ -613,5 +613,23 @@ def build_and_validate_article(*args)
...
   end
 
+  describe ".feed" do
+    it "returns published articles ordered by most recent published_at" do
+      first_article = create(:article, published: true, published_at: 2.hours.ago)
+      second_article = create(:article, published: true, published_at: 1.hour.ago)
+      create(:article, published: false)
+
+      expect(described_class.feed.pluck(:id)).to eq([second_article.id, first_article.id])
+    end
```
Comment (rhymes) [inline spec/models/article_spec.rb]: this is more a controller test, see the doubt about adding order to scopes

File: app/models/article.rb (hunk 1 above)
Comment (bradpauly) [reply app/models/article.rb]: It could definitely be a scope. I don't have a preference on ordering. I like to try keeping query specific info out of controllers. Let me know which you prefer and I'll make adjustments.

File: spec/models/article_spec.rb (hunk 4 above)
Comment (bradpauly) [reply spec/models/article_spec.rb]: I like to have query related tests in the model, but per other comment happy to move if desired.

**Comment by bradpauly** [comment]
@rhymes Addressed your feedback, let me know if there's anything else that needs to be changed.

**Comment by bradpauly** [comment]
@rhymes sorry, I misunderstood your comment. I assumed it was covered since this doesn't change behavior. I'm afraid I don't know how to test this in a request spec. They seem to be about the presence of content in the response body and not the structure of the content. I did a quick search to see what I could learn and didn't find a useful example. Is there a resource you could point me to that would help me figure it out?

**Comment by rhymes** [comment]
Hi @bradpauly, you're right, we should probably leave it like this. It might be done in a system test with some magic (because testing order isn't easy) instead of a request spec but I wouldn't hold off this PR for that.
